### ISSUE TITLE
[1.2.4] Automation Rules Broken for Data Queries

### ISSUE THREAD

**Issue by eschoeller** [Cacti/cacti#2774]
**Describe the bug**
After working through issue #2663 several of my graph automation rules have stopped working.

**To Reproduce**
1. Create a new device that has graph rules associated with it.
2. Observe that the graph rules are not creating graphs anymore.

**Expected behavior**
Automation rules that were working prior to fixes in #2663 (or something similar) should still be working.

**Screenshots**
Yes, I'll include some.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser chrome
 - Version n/a

**Comment by eschoeller** [comment]
What I did was change the device template for a particular device. It brought in the new data queries and ran them. But the graphs were never created. Just now I re-applied the graph automation rules. Things started to take shape:
```
2019/06/27 10:21:56 - DSTRACE Running Re-Index for Device[724], DQ[83]
2019/06/27 10:22:10 - DSTRACE Running Re-Index for Device[724], DQ[84]
2019/06/27 10:22:18 - DSTRACE Running Re-Index for Device[724], DQ[85]
2019/06/27 10:22:23 - DSTRACE Running Re-Index for Device[724], DQ[86]
2019/06/27 10:22:32 - DSTRACE Running Re-Index for Device[724], DQ[87]
2019/06/27 10:22:47 - DSTRACE Running Re-Index for Device[724], DQ[88]
2019/06/27 10:22:48 - DSTRACE Running Re-Index for Device[724], DQ[89]
2019/06/27 10:22:48 - DSTRACE Running Re-Index for Device[724], DQ[90]
2019/06/27 10:22:49 - DSTRACE Running Re-Index for Device[724], DQ[91]
2019/06/27 10:22:49 - DSTRACE Running Re-Index for Device[724], DQ[92]
2019/06/27 10:33:54 - DSTRACE Data Source Exists NOT Special Case "CordActivePower"
2019/06/27 10:33:54 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:54 - DSTRACE Data Source keys ""
2019/06/27 10:33:54 - DSTRACE Data Source values ""
2019/06/27 10:33:55 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67222], DS[70642, 70642]
2019/06/27 10:33:55 - DSTRACE Data Source Exists NOT Special Case "CordActivePower"
2019/06/27 10:33:55 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:55 - DSTRACE Data Source keys ""
2019/06/27 10:33:55 - DSTRACE Data Source values ""
2019/06/27 10:33:56 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67223], DS[70643, 70643]
2019/06/27 10:33:56 - DSTRACE Data Source Exists NOT Special Case "CordApparentPower"
2019/06/27 10:33:56 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:56 - DSTRACE Data Source keys ""
2019/06/27 10:33:56 - DSTRACE Data Source values ""
2019/06/27 10:33:57 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67224], DS[70644, 70644]
2019/06/27 10:33:57 - DSTRACE Data Source Exists NOT Special Case "CordApparentPower"
2019/06/27 10:33:57 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:57 - DSTRACE Data Source keys ""
2019/06/27 10:33:57 - DSTRACE Data Source values ""
2019/06/27 10:33:57 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67225], DS[70645, 70645]
2019/06/27 10:33:57 - DSTRACE Data Source Exists NOT Special Case "CordEnergy"
2019/06/27 10:33:57 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:57 - DSTRACE Data Source keys ""
2019/06/27 10:33:57 - DSTRACE Data Source values ""
2019/06/27 10:33:58 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67226], DS[70646, 70646]
2019/06/27 10:33:58 - DSTRACE Data Source Exists NOT Special Case "CordEnergy"
2019/06/27 10:33:58 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:58 - DSTRACE Data Source keys ""
2019/06/27 10:33:58 - DSTRACE Data Source values ""
2019/06/27 10:33:59 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67227], DS[70647, 70647]
2019/06/27 10:33:59 - DSTRACE Data Source Exists NOT Special Case "CordFrequency"
2019/06/27 10:33:59 - DSTRACE Data Source previous does not exist
2019/06/27 10:33:59 - DSTRACE Data Source keys ""
2019/06/27 10:33:59 - DSTRACE Data Source values ""
2019/06/27 10:34:00 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67228], DS[70648, 70648]
2019/06/27 10:34:00 - DSTRACE Data Source Exists NOT Special Case "CordFrequency"
2019/06/27 10:34:00 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:00 - DSTRACE Data Source keys ""
2019/06/27 10:34:00 - DSTRACE Data Source values ""
2019/06/27 10:34:01 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67229], DS[70649, 70649]
2019/06/27 10:34:01 - DSTRACE Data Source Exists NOT Special Case "CordOutOfBalance"
2019/06/27 10:34:01 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:01 - DSTRACE Data Source keys ""
2019/06/27 10:34:01 - DSTRACE Data Source values ""
2019/06/27 10:34:02 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67230], DS[70650, 70650]
2019/06/27 10:34:02 - DSTRACE Data Source Exists NOT Special Case "CordOutOfBalance"
2019/06/27 10:34:02 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:02 - DSTRACE Data Source keys ""
2019/06/27 10:34:02 - DSTRACE Data Source values ""
2019/06/27 10:34:03 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67231], DS[70651, 70651]
2019/06/27 10:34:03 - DSTRACE Data Source Exists NOT Special Case "CordPowerUtilized"
2019/06/27 10:34:03 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:03 - DSTRACE Data Source keys ""
2019/06/27 10:34:03 - DSTRACE Data Source values ""
2019/06/27 10:34:04 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67232], DS[70652, 70652]
2019/06/27 10:34:04 - DSTRACE Data Source Exists NOT Special Case "CordPowerUtilized"
2019/06/27 10:34:04 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:04 - DSTRACE Data Source keys ""
2019/06/27 10:34:04 - DSTRACE Data Source values ""
2019/06/27 10:34:05 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67233], DS[70653, 70653]
2019/06/27 10:34:05 - DSTRACE Data Source Exists NOT Special Case "CordPowerFactor"
2019/06/27 10:34:05 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:05 - DSTRACE Data Source keys ""
2019/06/27 10:34:05 - DSTRACE Data Source values ""
2019/06/27 10:34:07 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67234], DS[70654, 70654]
2019/06/27 10:34:07 - DSTRACE Data Source Exists NOT Special Case "CordPowerFactor"
2019/06/27 10:34:07 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:07 - DSTRACE Data Source keys ""
2019/06/27 10:34:07 - DSTRACE Data Source values ""
2019/06/27 10:34:08 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67235], DS[70655, 70655]
2019/06/27 10:34:08 - DSTRACE Data Source Exists NOT Special Case "LineCurrentUtil1,LineCurrentUtil2,LineCurrentUtil3"
2019/06/27 10:34:08 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:08 - DSTRACE Data Source keys ""
2019/06/27 10:34:08 - DSTRACE Data Source values ""
2019/06/27 10:34:09 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67236], DS[70656, 70656]
2019/06/27 10:34:09 - DSTRACE Data Source Exists NOT Special Case "LineCurrentUtil1,LineCurrentUtil2,LineCurrentUtil3"
2019/06/27 10:34:09 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:09 - DSTRACE Data Source keys ""
2019/06/27 10:34:09 - DSTRACE Data Source values ""
2019/06/27 10:34:10 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67237], DS[70657, 70657]
2019/06/27 10:34:11 - DSTRACE Data Source Exists NOT Special Case "LineCurrent1,LineCurrent2,LineCurrent3"
2019/06/27 10:34:11 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:11 - DSTRACE Data Source keys ""
2019/06/27 10:34:11 - DSTRACE Data Source values ""
2019/06/27 10:34:12 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67238], DS[70658, 70658]
2019/06/27 10:34:12 - DSTRACE Data Source Exists NOT Special Case "LineCurrent1,LineCurrent2,LineCurrent3"
2019/06/27 10:34:12 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:12 - DSTRACE Data Source keys ""
2019/06/27 10:34:12 - DSTRACE Data Source values ""
2019/06/27 10:34:13 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67239], DS[70659, 70659]
2019/06/27 10:34:13 - DSTRACE Data Source Exists NOT Special Case "LineCurrent4"
2019/06/27 10:34:13 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:13 - DSTRACE Data Source keys ""
2019/06/27 10:34:13 - DSTRACE Data Source values ""
2019/06/27 10:34:14 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67240], DS[70660, 70660]
2019/06/27 10:34:14 - DSTRACE Data Source Exists NOT Special Case "LineCurrent4"
2019/06/27 10:34:14 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:14 - DSTRACE Data Source keys ""
2019/06/27 10:34:14 - DSTRACE Data Source values ""
2019/06/27 10:34:15 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67241], DS[70661, 70661]
2019/06/27 10:34:15 - DSTRACE Data Source Exists NOT Special Case "LineCurrent4"
2019/06/27 10:34:15 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:15 - DSTRACE Data Source keys ""
2019/06/27 10:34:15 - DSTRACE Data Source values ""
2019/06/27 10:34:16 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67242], DS[70662, 70662]
2019/06/27 10:34:16 - DSTRACE Data Source Exists NOT Special Case "LineCurrent4"
2019/06/27 10:34:16 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:16 - DSTRACE Data Source keys ""
2019/06/27 10:34:16 - DSTRACE Data Source values ""
2019/06/27 10:34:16 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67243], DS[70663, 70663]
2019/06/27 10:34:17 - DSTRACE Data Source Exists NOT Special Case "PhaseActivePower1,PhaseActivePower2,PhaseActivePower3"
2019/06/27 10:34:17 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:17 - DSTRACE Data Source keys ""
2019/06/27 10:34:17 - DSTRACE Data Source values ""
2019/06/27 10:34:17 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67244], DS[70664, 70664]
2019/06/27 10:34:18 - DSTRACE Data Source Exists NOT Special Case "PhaseActivePower1,PhaseActivePower2,PhaseActivePower3"
2019/06/27 10:34:18 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:18 - DSTRACE Data Source keys ""
2019/06/27 10:34:18 - DSTRACE Data Source values ""
2019/06/27 10:34:19 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67245], DS[70665, 70665]
2019/06/27 10:34:19 - DSTRACE Data Source Exists NOT Special Case "PhaseApparentPower1,PhaseApparentPower2,PhaseApparentPower3"
2019/06/27 10:34:19 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:19 - DSTRACE Data Source keys ""
2019/06/27 10:34:19 - DSTRACE Data Source values ""
2019/06/27 10:34:20 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67246], DS[70666, 70666]
2019/06/27 10:34:20 - DSTRACE Data Source Exists NOT Special Case "PhaseApparentPower1,PhaseApparentPower2,PhaseApparentPower3"
2019/06/27 10:34:20 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:20 - DSTRACE Data Source keys ""
2019/06/27 10:34:20 - DSTRACE Data Source values ""
2019/06/27 10:34:21 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67247], DS[70667, 70667]
2019/06/27 10:34:21 - DSTRACE Data Source Exists NOT Special Case "PhaseCrestFactor1,PhaseCrestFactor2,PhaseCrestFactor3"
2019/06/27 10:34:21 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:21 - DSTRACE Data Source keys ""
2019/06/27 10:34:21 - DSTRACE Data Source values ""
2019/06/27 10:34:22 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67248], DS[70668, 70668]
2019/06/27 10:34:22 - DSTRACE Data Source Exists NOT Special Case "PhaseCrestFactor1,PhaseCrestFactor2,PhaseCrestFactor3"
2019/06/27 10:34:22 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:22 - DSTRACE Data Source keys ""
2019/06/27 10:34:22 - DSTRACE Data Source values ""
2019/06/27 10:34:23 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67249], DS[70669, 70669]
2019/06/27 10:34:23 - DSTRACE Data Source Exists NOT Special Case "PhaseCurrent1,PhaseCurrent2,PhaseCurrent3"
2019/06/27 10:34:23 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:23 - DSTRACE Data Source keys ""
2019/06/27 10:34:23 - DSTRACE Data Source values ""
2019/06/27 10:34:24 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67250], DS[70670, 70670]
2019/06/27 10:34:24 - DSTRACE Data Source Exists NOT Special Case "PhaseCurrent1,PhaseCurrent2,PhaseCurrent3"
2019/06/27 10:34:24 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:24 - DSTRACE Data Source keys ""
2019/06/27 10:34:24 - DSTRACE Data Source values ""
2019/06/27 10:34:25 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67251], DS[70671, 70671]
2019/06/27 10:34:25 - DSTRACE Data Source Exists NOT Special Case "PhaseEnergy1,PhaseEnergy2,PhaseEnergy3"
2019/06/27 10:34:25 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:25 - DSTRACE Data Source keys ""
2019/06/27 10:34:25 - DSTRACE Data Source values ""
2019/06/27 10:34:26 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67252], DS[70672, 70672]
2019/06/27 10:34:26 - DSTRACE Data Source Exists NOT Special Case "PhaseEnergy1,PhaseEnergy2,PhaseEnergy3"
2019/06/27 10:34:26 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:26 - DSTRACE Data Source keys ""
2019/06/27 10:34:26 - DSTRACE Data Source values ""
2019/06/27 10:34:27 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67253], DS[70673, 70673]
2019/06/27 10:34:27 - DSTRACE Data Source Exists NOT Special Case "PhasePowerFactor1,PhasePowerFactor2,PhasePowerFactor3"
2019/06/27 10:34:27 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:27 - DSTRACE Data Source keys ""
2019/06/27 10:34:27 - DSTRACE Data Source values ""
2019/06/27 10:34:28 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67254], DS[70674, 70674]
2019/06/27 10:34:28 - DSTRACE Data Source Exists NOT Special Case "PhasePowerFactor1,PhasePowerFactor2,PhasePowerFactor3"
2019/06/27 10:34:28 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:28 - DSTRACE Data Source keys ""
2019/06/27 10:34:28 - DSTRACE Data Source values ""
2019/06/27 10:34:29 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67255], DS[70675, 70675]
2019/06/27 10:34:29 - DSTRACE Data Source Exists NOT Special Case "PhaseVoltage1,PhaseVoltage2,PhaseVoltage3"
2019/06/27 10:34:29 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:29 - DSTRACE Data Source keys ""
2019/06/27 10:34:29 - DSTRACE Data Source values ""
2019/06/27 10:34:30 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67256], DS[70676, 70676]
2019/06/27 10:34:30 - DSTRACE Data Source Exists NOT Special Case "PhaseVoltage1,PhaseVoltage2,PhaseVoltage3"
2019/06/27 10:34:30 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:30 - DSTRACE Data Source keys ""
2019/06/27 10:34:30 - DSTRACE Data Source values ""
2019/06/27 10:34:31 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67257], DS[70677, 70677]
2019/06/27 10:34:31 - DSTRACE Data Source Exists NOT Special Case "PhaseVoltageDev1,PhaseVoltageDev2,PhaseVoltageDev3"
2019/06/27 10:34:31 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:31 - DSTRACE Data Source keys ""
2019/06/27 10:34:31 - DSTRACE Data Source values ""
2019/06/27 10:34:32 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67258], DS[70678, 70678]
2019/06/27 10:34:32 - DSTRACE Data Source Exists NOT Special Case "PhaseVoltageDev1,PhaseVoltageDev2,PhaseVoltageDev3"
2019/06/27 10:34:32 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:32 - DSTRACE Data Source keys ""
2019/06/27 10:34:32 - DSTRACE Data Source values ""
2019/06/27 10:34:33 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67259], DS[70679, 70679]
2019/06/27 10:34:34 - DSTRACE Data Source Exists NOT Special Case "BranchCurrent1,BranchCurrent2,BranchCurrent3,BranchCurrent4,BranchCurrent5,BranchCurrent6"
2019/06/27 10:34:34 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:34 - DSTRACE Data Source keys ""
2019/06/27 10:34:34 - DSTRACE Data Source values ""
2019/06/27 10:34:35 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67260], DS[70680, 70680]
2019/06/27 10:34:35 - DSTRACE Data Source Exists NOT Special Case "BranchCurrent1,BranchCurrent2,BranchCurrent3,BranchCurrent4,BranchCurrent5,BranchCurrent6"
2019/06/27 10:34:35 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:35 - DSTRACE Data Source keys ""
2019/06/27 10:34:35 - DSTRACE Data Source values ""
2019/06/27 10:34:36 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67261], DS[70681, 70681]
2019/06/27 10:34:36 - DSTRACE Data Source Exists NOT Special Case "BranchCurrentUtil1,BranchCurrentUtil2,BranchCurrentUtil3,BranchCurrentUtil4,BranchCurrentUtil5,BranchCurrentUtil6"
2019/06/27 10:34:36 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:36 - DSTRACE Data Source keys ""
2019/06/27 10:34:36 - DSTRACE Data Source values ""
2019/06/27 10:34:37 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67262], DS[70682, 70682]
2019/06/27 10:34:37 - DSTRACE Data Source Exists NOT Special Case "BranchCurrentUtil1,BranchCurrentUtil2,BranchCurrentUtil3,BranchCurrentUtil4,BranchCurrentUtil5,BranchCurrentUtil6"
2019/06/27 10:34:37 - DSTRACE Data Source previous does not exist
2019/06/27 10:34:37 - DSTRACE Data Source keys ""
2019/06/27 10:34:37 - DSTRACE Data Source values ""
2019/06/27 10:34:38 - AUTOM8 NOTE: Graph Added - Device[724], Graph[67263], DS[70683, 70683]
2019/06/27 10:34:38 - AUTOM8 WARNING: Automation Rule[Server Technology CDU - Environmental Monitors (C) - Sentry4] for Device[724] - DQ[90] includes a SQL column tempHumidSensorStatus that is not found for the Device.  Can not continue.
2019/06/27 10:34:38 - AUTOM8 WARNING: Automation Rule[Server Technology CDU - Environmental Monitors (D) - Sentry4] for Device[724] - DQ[91] includes a SQL column tempHumidSensorStatus that is not found for the Device.  Can not continue.
```
I now have _most_ of the graphs I needed, except for some temperature graphs, but I know why that isn't working, it's related to #2658 so I have to create those graphs manually until that other issue is resolved. 

So, in short, for some reason when adding a data query to an existing device the AUTOM8 rules aren't running correctly.

**Comment by cigamit** [comment]
@eschoeller, edit each of your Graph Automation rules, and ensure that they are still mapped correctly.  I just discovered, on new installs, that the Graph Automation rules are scrambled.  It's a design issue that I'm discussing with @netniV at the moment.

**Comment by cigamit** [comment]
This is a duplicate of https://github.com/Cacti/cacti/issues/2775

**Comment by cigamit** [comment]
Marking this one duplicate.
